lib/explorer/dashboard.ex

```diff
@@ -49,15 +51,24 @@ defmodule Explorer.Dashboard do
       issuer: block["issuer"],
       tx_count: Enum.count(block["transactions"]),
       ada_output: ada_output,
-      fees: fees
+      fees: fees,
+      date_time: date_time_utc()
     }
 
     all_blocks = BlocksDb.get_all_blocks()
-    # First, get the oldest / @max_blocks_per_page-nth block on the db
+    # Get the @max_blocks_per_page-nth block on the db
+    # so it can be removed from the table display. This might
+    # be nil while BlockDb is still building up to @max_blocks_per_page
```

Comment:
@caike did you consider setting a [`:limit`](https://hexdocs.pm/phoenix_live_view/Phoenix.LiveView.html#stream/4:~:text=%3Alimit%20%2D%20the%20optional%20positive%20or%20negative%20number%20of%20results%20to%20limit%20on%20the%20UI%20on%20the%20client.%20As%20new%20items%20are%20streamed%2C%20the%20UI%20will%20remove%20existing%20items%20to%20maintain%20the%20limit.%20For%20example%2C%20to%20limit%20the%20stream%20to%20the%20last%2010%20items%20in%20the%20UI%20while%20appending%20new%20items%2C%20pass%20a%20negative%20value%3A) on the stream? I wonder if that would eliminate the code paths dealing with `block_to_be_removed`.

If you tried it, did you run into any problems? I have seen a few stream-related bugs, but they've been gradually fixed in latest LV versions.

Reply:
Good call! Looks much better https://github.com/wowica/blocks/pull/7